xpk.py

```diff
@@ -820,6 +844,9 @@ def get_capacity_arguments(args) -> tuple[str, int]:
     capacity_args = ""
     num_types+=1
   if args.reservation:
+    return_code = verify_reservation_exists(args)
+    if return_code > 0:
```

Comment:
should we warn if we're using on-demand?

Reply:
Hmm I think we don't need to since the user would have to specify `--on-demand` in their command.

Reply:
Fair.